### PR TITLE
GHA vulnerability scan for Golang

### PR DESCRIPTION
**Pull request by nextwebb**
Vulnerability scan  workflow for Golang using Snyk #94

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+jobs:
+   security::
+    runs-on: ubuntu-latest
+    strategy:
+       matrix:
+         go-version: [1.13, 1.14, 1.15, 1.x]
```
Comment (adefemi171) [inline .github/workflows/main.yml]: I think we can stick with one version of go, you can check other workflows to compare

File: .github/workflows/main.yml [hunk 2]
```diff
@@ -0,0 +1,29 @@
+
+name: Backend workflow for Golang using Snyk
+
+on:
+  push:
```
Comment (adefemi171) [inline .github/workflows/main.yml:37]: I think the job should run on pull request, this can be removed

File: .github/workflows/main.yml [hunk 3]
```diff
@@ -0,0 +1,29 @@
+
+name: Backend workflow for Golang using Snyk
```
Comment (adefemi171) [inline .github/workflows/main.yml]: You can leave the name short, like **Vulnerability Scan**

File: .github/workflows/main.yml (hunk 2 above)
Comment (nextwebb) [reply .github/workflows/main.yml:37]: okay

File: .github/workflows/main.yml (hunk 3 above)
Comment (nextwebb) [reply .github/workflows/main.yml]: okay

File: .github/workflows/main.yml (hunk 1 above)
Comment (nextwebb) [reply .github/workflows/main.yml]: alright. thanks

**Pull request closed without merging**